packages/hdwallet-tallyho/src/tallyho.ts

```diff
@@ -214,6 +262,53 @@ export class TallyHoHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  private async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
```

Comment:
Note: We might get away with a lighter implementation since the checks for being in the context of a Tally provider should be completed, but IMO doesn't hurt to err on the side of safety and keep the full detection implementation